wpilibj/src/test/java/edu/wpi/first/wpilibj/event/BooleanEventTest.java

```diff
@@ -31,42 +31,37 @@ void testBinaryCompositions() {
   }
 
   @Test
-  void testBinaryCompositionLoopSemantics() {
-    var loop1 = new EventLoop();
-    var loop2 = new EventLoop();
-
+  void testLoopValueSemantics() {
+    var loop = new EventLoop();
+    var bool1 = new AtomicBoolean(true);
+    var bool2 = new AtomicBoolean(true);
     var counter1 = new AtomicInteger(0);
     var counter2 = new AtomicInteger(0);
 
-    new BooleanEvent(loop1, () -> true)
-        .and(new BooleanEvent(loop2, () -> true))
+    new BooleanEvent(
+            loop,
+            () -> {
+              bool2.set(false);
+              return bool1.get();
+            })
+        .and(new BooleanEvent(loop, bool2::get))
         .ifHigh(counter1::incrementAndGet);
 
-    new BooleanEvent(loop2, () -> true)
-        .and(new BooleanEvent(loop1, () -> true))
-        .ifHigh(counter2::incrementAndGet);
-
     assertEquals(0, counter1.get());
     assertEquals(0, counter2.get());
 
-    loop1.poll();
+    loop.poll();
 
-    assertEquals(1, counter1.get());
+    assertEquals(0, counter1.get());
     assertEquals(0, counter2.get());
-
-    loop2.poll();
-
-    assertEquals(1, counter1.get());
-    assertEquals(1, counter2.get());
   }
 
   @Test
   void testEdgeDecorators() {
+    var loop = new EventLoop();
     var bool = new AtomicBoolean(false);
     var counter = new AtomicInteger(0);
 
-    var loop = new EventLoop();
-
```

Comment:
This seems like a redundant change.

Reply:
It is, but all the other tests create the loop first, so this is for consistency.